discord/message.py

```diff
@@ -452,6 +455,92 @@ def guild_id(self) -> Optional[int]:
         return self._parent.guild_id
 
 
+class MessageSnapshot:
```

Comment:
I think this could have a ``cached_message`` attribute that, as its name indicates, has the cached message of this snapshot, if found.

Reply:
Is it ever found? I tested it for a while and none of the forwards I saw had that attribute non-empty

Reply:
As said, if the message is found in the cache then this would be non-empty. If a user forwards a message the bot has cached then the attribute would be filled, else not.

Reply:
Wait sorry I misunderstood. Yeah that seems like not a bad idea, though I'd need to look into how the cache works exactly

Reply:
This is a good idea, it shouldn't be a complicated add either. You already pass the connection state as a parameter but you need to bind it to make it into a property.